### PR TITLE
docs: add new features to the policy-controller

### PR DESCRIPTION
**Pull request by hectorj2f**
Signed-off-by: Hector Fernandez <hector@chainguard.dev>

#### Summary

Update the policy-controller documentation with the new features.

#### Release Note

Update the documentation of the policy-controller.

### CODE REVIEW

File: content/en/policy-controller/overview.md [hunk 1]
```diff
@@ -119,6 +116,7 @@ spec:
 
 Each `key` authority can contain these properties:
 - `key.data`: specifies the plain text string of the public key
+- `key.hashAlgorithm` (optional): specifies the signature digest for the key, e.g. `sha512`, `sha256`, `sha384` or `sha224`
```
Comment (vaikas) [inline content/en/policy-controller/overview.md:119]: since this is optional, might want to mention the default.

**Review by vaikas — COMMENTED** [review]
We should also remove this section now:
https://docs.sigstore.dev/policy-controller/overview#deprecated-policy-controller-validation-behavior

or basically here:
https://github.com/sigstore/docs/blob/main/content/en/policy-controller/overview.md#deprecated-policy-controller-validation-behavior

File: content/en/policy-controller/overview.md (hunk 1 above)
Comment (hectorj2f) [reply content/en/policy-controller/overview.md:119]: yes, I removed 
https://docs.sigstore.dev/policy-controller/overview#deprecated-policy-controller-validation-behavior

**Review by vaikas — APPROVED** [review]
Thank you!!!

**Comment by hectorj2f** [comment]
@ltagliaferri Please, could you merge this PR 🙏🏻 ?

**Review by ltagliaferri — APPROVED** [review]

**Pull request merged**